packages/eslint-config-ckeditor5/eslint.config.mjs

```diff
@@ -0,0 +1,491 @@
+
+			'@stylistic/arrow-spacing': 'error',
+
+			'@stylistic/generator-star-spacing': [ 'error', 'after' ],
+
+			'@stylistic/template-curly-spacing': [ 'error', 'always' ],
+
+			'@stylistic/yield-star-spacing': [ 'error', 'after' ]
+
+			// This is too pedantic. It makes writing callbacks such as `value => value ? 1 : 2` impossible
+			// '@stylistic/no-confusing-arrow': 'error',
+
+			// We can't use this rule because we allow comments before `else`.
+			// Make sure to report such case to ESLint.
+			// '@stylistic/brace-style': [
+			// 	'error',
+			// 	'1tbs'
+			// ],
+		}
+	}
+];
+
+const rulesTypeScript = [
+	{
+		plugins: {
+			ts
+		},
+
+		extends: [ 'ts/recommended' ],
+
+		files: [ '**/*.@(ts|tsx)' ],
+
+		rules: {
+			/*
+			╭──────────────────────────────────────────────────────────────────────╮
+			│ Rules for detecting errors, problems, suggestions and best practices │
+			╰──────────────────────────────────────────────────────────────────────╯
+			*/
+
+			'@typescript-eslint/array-type': [ 'error', {
+				default: 'generic'
+			} ],
+
+			'@typescript-eslint/no-unsafe-function-type': 'off',
+
+			'@typescript-eslint/no-empty-object-type': 'error',
+
+			'@typescript-eslint/no-wrapper-object-types': 'error',
+
+			'@typescript-eslint/consistent-type-assertions': [ 'error', {
+				assertionStyle: 'as',
+				objectLiteralTypeAssertions: 'allow-as-parameter'
+			} ],
+
+			'@typescript-eslint/consistent-type-imports': 'error',
+
+			'@typescript-eslint/explicit-module-boundary-types': [ 'error', {
+				allowedNames: METHODS_THAT_USE_AS_CONST_INSTEAD_OF_RETURN_TYPE,
+				allowArgumentsExplicitlyTypedAsAny: true
+			} ],
+
+			'@typescript-eslint/explicit-member-accessibility': [ 'error', {
+				accessibility: 'explicit',
+				overrides: {
+					constructors: 'off'
+				}
+			} ],
+
+			'@typescript-eslint/no-confusing-non-null-assertion': 'error',
+
+			'@typescript-eslint/no-empty-function': 'off',
+
+			'@typescript-eslint/no-empty-interface': 'off',
+
+			'@typescript-eslint/no-explicit-any': 'off',
+
+			'@typescript-eslint/no-inferrable-types': 'off',
+
+			'@typescript-eslint/no-invalid-void-type': 'error',
+
+			'@typescript-eslint/no-non-null-asserted-nullish-coalescing': 'error',
+
+			'@typescript-eslint/no-non-null-assertion': 'off',
+
+			'@typescript-eslint/parameter-properties': 'error',
+
+			'@typescript-eslint/unified-signatures': 'error',
+
+			/*
+			╭─────────────────╮
+			│ Extension rules │
+			╰─────────────────╯
+			*/
+
+			'no-unused-expressions': 'off',
+			'@typescript-eslint/no-unused-expressions': 'error',
+
+			'no-unused-vars': 'off',
+			'@typescript-eslint/no-unused-vars': 'error',
+
+			'no-use-before-define': 'off',
+			'@typescript-eslint/no-use-before-define': [ 'error', {
+				functions: false,
+				classes: false,
+				variables: true,
+				typedefs: false,
+				ignoreTypeReferences: true
+			} ],
+
+			'no-array-constructor': 'off',
+			'@typescript-eslint/no-array-constructor': 'error',
```

Comment:
According to https://eslint.org/docs/latest/rules/no-array-constructor:

> This rule additionally supports TypeScript type syntax.

I wanted to verify what it means. I created two TypeScript files:

* `ckeditor5-linters-config/packages/eslint-config-ckeditor5/customfile.ts`
* `ckeditor5-linters-config/packages/eslint-plugin-ckeditor5-rules/customfileinplugin.ts`

With the following content:

```ts
Array();

Array(0, 1, 2);

new Array(0, 1, 2);

Array(...args);
```

Then I updated the `yarn lint` script: `"lint": "eslint --quiet \"**/*.{js,mjs|cjs|ts|mts|cts}\"",`.

Unfortunately, `yarn lint` does not detect anything wrong in both files.

Reply:
Try changing the `yarn lint` script to `"lint": "eslint --quiet \"**/*.{js,mjs,cjs,ts,mts,cts}\"",` and check again 😄